src/index.ts

```diff
@@ -26,13 +26,19 @@ function loader(this: interfaces.Webpack, contents: string) {
         return;
     }
 
+    // Load files generated by previous loaders in the same chain
+    const generatedFiles = getGeneratedFiles(this);
+    for (let filePath of Object.keys(generatedFiles)) {
+        updateFileInCache(filePath, generatedFiles[filePath], instance);
+    }
+
     const rawFilePath = path.normalize(this.resourcePath);
     const filePath = utils.appendTsSuffixIfMatch(options.appendTsSuffixTo, rawFilePath);
     const fileVersion = updateFileInCache(filePath, contents, instance);
 
     const { outputText, sourceMapText } = options.transpileOnly
         ? getTranspilationEmit(filePath, contents, instance, this)
```

Comment:
Filtering of dependencies has been moved to new function `removeGeneratedFileDependencies` in the latest commit. The call to this function is, of course, only made if the `usePreviousLoaderGeneratedFiles` option is true.